source/extensions/common/wasm/foreign.cc

```diff
@@ -203,16 +203,23 @@ class CreateExpressionFactory : public ExpressionFactory {
       auto token = expr_context.createToken();
       auto& handler = expr_context.getExpression(token);
 
-      handler.parsed_expr_ = parse_status.value();
-      auto cel_expression_status = expr_context.builder()->CreateExpression(
-          &handler.parsed_expr_.expr(), &handler.parsed_expr_.source_info());
+      const auto& parsed_expr = parse_status.value();
+      const cel::expr::Expr& cel_expr = parsed_expr.expr();
+      const cel::expr::SourceInfo& cel_source_info = parsed_expr.source_info();
+
+      std::vector<absl::Status> warnings;
+      auto cel_expression_status =
+          expr_context.builder()->CreateExpression(&cel_expr, &cel_source_info, &warnings);
+
       if (!cel_expression_status.ok()) {
         ENVOY_LOG(info, "expr_create compile error: {}", cel_expression_status.status().message());
         expr_context.deleteExpression(token);
         return WasmResult::BadArgument;
       }
 
       handler.compiled_expr_ = std::move(cel_expression_status.value());
+      handler.parsed_expr_ = parsed_expr;
```

Comment:
This will copy parsed_expr and delete it, leaving the expression points dangling... Please don't re-order statements.